### PR TITLE
Run ember-cli-update to v3.28.3 to align with the latest addon blueprint

### PR DESCRIPTION
**Pull request by SergeAstapov**
This PR updates addon setup to align with the latest blueprint in Ember CLI to simplify further updates.

Most notable, new ember-try scenarios added to CI:
* embroider-safe
* embroider-optimized

To reduce amount of changes in this PR, come changes moved to separate PRs:
* Drop old Node.js versions https://github.com/tracked-tools/tracked-toolbox/pull/44
* npmignore file updates https://github.com/tracked-tools/tracked-toolbox/pull/43
* Switch `ember-cli-eslint` -> `eslint` https://github.com/tracked-tools/tracked-toolbox/pull/45
* prettier introduced in #46
* eslint-plugin-qunit introduced in #49

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,9 +29,9 @@ jobs:
       - name: Install Dependencies
         run: yarn install --frozen-lockfile
       - name: Lint
-        run: yarn lint:js
+        run: yarn lint
```
Comment (SergeAstapov) [inline .github/workflows/ci.yml:32]: new we lint templates as well (although hbs exists in dummy app only)

File: README.md [hunk 2]
```diff
@@ -8,8 +8,8 @@ tracking!
 Compatibility
 ------------------------------------------------------------------------------
 
-* Ember.js v3.8 or above
-* Ember CLI v2.13 or above
+* Ember.js v3.20 or above
+* Ember CLI v3.20 or above
 * Node.js v8 or above
```
Comment (SergeAstapov) [inline README.md:13]: support node versions updated separately in #44

File: config/ember-try.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 const getChannelURL = require('ember-source-channel-url');
-// const { embroiderSafe, embroiderOptimized } = require('@embroider/test-setup');
+const { embroiderSafe, embroiderOptimized } = require('@embroider/test-setup');
```
Comment (SergeAstapov) [inline config/ember-try.js:4]: this now uncommented and pass (addon is simple so shouldn't fail at all)

**Comment by chriskrycho** [comment]
Thanks for this @SergeAstapov – mind rebasing? I'll land it after it's resolved

**Comment by SergeAstapov** [comment]
> Thanks for this @SergeAstapov – mind rebasing? I'll land it after it's resolved

done!

Once this is done, I'll create another PR to update blueprint to 4.2

**Pull request merged**